content/chapter-01/LabCloudVsOnPremise.md

```diff
@@ -0,0 +1,39 @@
+# Lab : Cloud vs. on-premise
+
+* Let us list a few activities we do on a [Software Development Life Cycle](https://en.wikipedia.org/wiki/Systems_development_life_cycle):
+
+  * Development platform (Ex. Java, .Net, Node.js, Python, etc.)
+  * Storage:
+    * Deployment articraft (Ex. .dlls, .war, .ear, files, etc)
+    * Data Storage (Ex. RDBMS, NoSQL Database, files)
+    * Source control (Ex. Git, SVN, Team Foundation Server)
+  * Non-functional requirements:
+    * Traffic estimation (Ex. request/second, jobs/minute)
+    * Response time (Ex. max response time [ms])
+    * Supported number of users
+    * Up-time requirements (Ex. Can we have maintenance windows? How many nines we need?)
+    * Disaster recovery plans (Ex. We need to store data for a year)
+    * Data locality (Ex. We need to be available worldwide but data is consumed locally, or a user in London should be able to access data from New York)
```

Comment:
Data locality, e.g.: we need to make the system available worldwide, but data is consumed locally. Should a user in London, UK be able to access data from New York, NY?

Reply:
Done